File: tableau/TableauMarchand.java
package tableau;

import deplacement.Deplacement;
import marchand.Marchand;

public class TableauMarchand extends Tableau
{
private Marchand marchand;
	
	public TableauMarchand(int X, int Y, Marchand marchand)
	{
		super(X, Y);
		this.setMarchand(marchand);
		
	}

	public Marchand getMarchand()
	{
		return marchand;
	}

	private void setMarchand(Marchand marchand)
	{
		this.marchand = marchand;
	}
	
	public void evenement()
	{
		Marchand.trade(this);
		Deplacement.deplacement(this);
	}
	
}

File: Cs/tableau/TableauMarchand.cs
using Deplacement;
using Marchand; 

namespace TableauNamespace
{
    public class TableauMarchand : Tableau
    {
        private Marchand marchand;

        public TableauMarchand(int x, int y, Marchand marchand) : base(x, y)
        {
            this.marchand = marchand;
        }

        public Marchand GetMarchand()
        {
            return marchand;
        }

        private void SetMarchand(Marchand marchand)
        {
            this.marchand = marchand;
        }

        public override void Evenement()
        {
            Marchand.Trade(this);
            Deplacement.Deplacement(this);
        }
    }
}
